### PR TITLE
Implement footer links and revised copyright

### PR DESCRIPTION
**Pull request by VirginiaDooley**
Ref: [DemocracyClub/design-system#140](https://github.com/DemocracyClub/design-system/issues/140)

Blocked by https://github.com/DemocracyClub/dc_django_utils/pull/91 and https://github.com/DemocracyClub/design-system/pull/143

Review at https://stage.democracyclub.org.uk/

This pull request implements a split footer that organizes footer links into grouped columns and includes a dedicated section for copyright and Companies House information. 

**Questions for Reviewers:**

- [x] Should the footer links in "Made by Democracy Club" direct to the corresponding project pages or external home pages?
- [x] Are there any other content changes needed for the menus?
- [x] What is the ideal mobile layout given that the mockup in the issue has fewer footer links that we would like have, at least on Website? 

TODO: 

- [x] Bump dc_utils
- [x] Bump design system
- [x] Deploy to staging

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1208578619254589

### CODE REVIEW

**Comment by pmk01** [comment]
This is great, thanks! My only immediate thought is that the mailing list link could be changed to link to the [archive](https://mailinglist.democracyclub.org.uk/archive), simply becuase we've already got a signup button. Oh and we should probably change Twitter to X.com...

**Comment by pmk01** [comment]
Also when I squash it down the header text becomes really small. Could it also maybe be centred?

Screenshot 2024-10-21 143833

**Comment by VirginiaDooley** [comment]
> This is great, thanks! My only immediate thought is that the mailing list link could be changed to link to the [archive](https://mailinglist.democracyclub.org.uk/archive), simply becuase we've already got a signup button. Oh and we should probably change Twitter to X.com...

So I presumed we remove the sign up button and add to menu. Do you think that's a bad idea? It's definitely not as eye catching to have as a footer link. Noted on Twitter.

**Comment by pmk01** [comment]
Oh, yes that's also fine, I think, but interested to know what @symroe thinks.

File: democracy_club/templates/base.html
Comment (symroe) [inline democracy_club/templates/base.html:76]: I'm not sure what's going on with the deployment, but I'm seeing `h6` tags there rather than this code.

Either way, I think the headings can be a bit larger, especially as Peter points out they actually go smaller than the normal font size at lower widths.

Maybe `ds-h5`?

**Review by symroe — APPROVED** [review]
I think this is good. Small point about the class and font size.

In the design system issue the background colour is different. Did you try and discard this?

**Comment by VirginiaDooley** [comment]
> I think this is good. Small point about the class and font size.
> 
> In the design system issue the background colour is different. Did you try and discard this?

I thought I would keep the change simple at this point, but happy to try this in https://github.com/DemocracyClub/design-system/pull/143 and come back here.

File: democracy_club/templates/base.html
Comment (VirginiaDooley) [reply democracy_club/templates/base.html]: @symroe thoughts on mailing list link v CTA?

**Comment by symroe** [comment]
Blocked on https://github.com/DemocracyClub/dc_django_utils/pull/91

File: .circleci/config.yml
Comment (chris48s) [inline .circleci/config.yml:153]: TODO: remove before merging

File: Pipfile.lock
Comment (chris48s) [inline Pipfile.lock:1249]: The reason why this PR is changing the version of unrelated package like ruff is probably because someone who worked on this PR (either you or Virginia - haven't checked the commits)

- edited `Pipfile`
- ran `pipenv lock`

This has the knock-on effect of updating every package in your tree which isn't pinned with a `==` in Pipfile.

The way to update only `dc-django-utils` without also changing the version of other packages where you've got stuff like `ruff = "*"` in the `Pipfile` would be to run `pipenv update "dc-django-utils@https://github.com/DemocracyClub/dc_django_utils/archive/refs tags/8.0.0.tar.gz"`.

In general, I am most concerned about spending our effort on moving to a tool that makes it easier to do the right thing by default (uv) rather than continuing to faff about with pipenv and bikeshed this on every PR, but seeing as it is something that has come up here I thought I'd mention it as there is probably still a long period of being stuck with pipenv on some subset of projects ahead of us.

Comment (symroe) [reply Pipfile.lock:1249]: Yeah, I ended up in a big merge conflict with `Pipefile.lock` and ended up having to re-lock. Not ideal. Do you have a good workflow for that?!

Comment (symroe) [reply Pipfile.lock]: Ok, I ran `pipenv upgrade` and still have changes in the lockfile that I don't really understand...

Comment (chris48s) [reply Pipfile.lock]: Sorry :facepalm: I didn't mean `pipenv upgrade`.

I mean your `pipenv update <package>` commands

So I'd accept the lockfile from `master` and then run

- `pipenv update "dc-django-utils@https://github.com/DemocracyClub/dc_django_utils/archive/refs tags/8.0.0.tar.gz"`
- `pipenv update "dc-design-system@https://github.com/DemocracyClub/design-system/archive/refs/tags/0.6.0.tar.gz"`

to apply the updates I want on top of it (even if those changes are already made in the `Pipfile` on my branch)

Comment (symroe) [reply Pipfile.lock]: Oh yeah, sorry, when I said above `I ran pipenv upgrade` I meant `pipenv upgrade [package]`. The committed pipenv lock is what is on master plus the changes after running the command you suggested

Comment (chris48s) [reply Pipfile.lock]: OK. 2 things:

1. In the original diff I looked at on this PR, you were upgrading design system to version 0.6.0. In the most recent force push, that change has been rolled back, even in the `Pipfile`

2. There are 2 commands in pipenv: `pipenv update` and `pipenv upgrade`. Despite having nearly identical names, they do different things. I have to look it up every time I attempt to use one. I have not helped this by using the wrong one in my initial post. Which one did you run? `update` or `upgrade`?

Comment (chris48s) [reply Pipfile.lock]: I think the diff you're trying to arrive at here should be:

```diff
diff --git a/Pipfile b/Pipfile
index 65b4bf3..1c528b6 100644
--- a/Pipfile
+++ b/Pipfile
@@ -11,7 +11,6 @@ django-localflavor = "==4.0"
 django-model-utils = "==5.0.0"
 django-storages = "==1.14.4"
 djhtml = "*"
-dc-design-system = {file = "https://github.com/DemocracyClub/design-system/archive/refs/tags/0.5.0.tar.gz"}
 django-typogrify = {file = "https://github.com/matthewn/django-typogrify/archive/refs/tags/v2.0.tar.gz"}
 jsonfield = "==3.1.0"
 markdown-headdown = "==0.1.3"
@@ -27,7 +26,8 @@ pre-commit = "*"
 django = "==4.2.16"
 zipp = "*"
 certifi = "*"
-dc-django-utils = {file = "https://github.com/DemocracyClub/dc_django_utils/archive/refs/tags/7.0.1.tar.gz"}
+dc-django-utils = {file = "https://github.com/DemocracyClub/dc_django_utils/archive/refs/tags/8.0.0.tar.gz"}
+dc-design-system = {file = "https://github.com/DemocracyClub/design-system/archive/refs/tags/0.6.0.tar.gz"}
 
 [dev-packages]
 vcrpy = "==6.0.2"
diff --git a/Pipfile.lock b/Pipfile.lock
index b3ac065..318c2da 100644
--- a/Pipfile.lock
+++ b/Pipfile.lock
@@ -1,7 +1,7 @@
 {
     "_meta": {
         "hash": {
-            "sha256": "86c89812ecdc42bcd78b508e41caff355b0619d6f880b77159188a27871994c5"
+            "sha256": "fba1ba0f4200a107e65cbddd8bc344da509b3eef753ad49c1f68bdb42e9c8b09"
         },
         "pipfile-spec": 6,
         "requires": {
@@ -67,14 +67,10 @@
             "version": "==3.4.0"
         },
         "dc-design-system": {
-            "file": "https://github.com/DemocracyClub/design-system/archive/refs/tags/0.5.0.tar.gz"
+            "file": "https://github.com/DemocracyClub/design-system/archive/refs/tags/0.6.0.tar.gz"
         },
         "dc-django-utils": {
-            "file": "https://github.com/DemocracyClub/dc_django_utils/archive/refs/tags/7.0.1.tar.gz",
-            "hashes": [
-                "sha256:06301ee79aa1d55da9e4aeeb8f750c43cbb1c59d7a53dc6550fad04737d6b68a"
-            ],
-            "version": "==7.0.1"
+            "file": "https://github.com/DemocracyClub/dc_django_utils/archive/refs/tags/8.0.0.tar.gz"
         },
         "distlib": {
             "hashes": [
@@ -225,6 +221,7 @@
                 "sha256:f5da449a6e1c989a4cea2631aa8ee67caa5a2ef855d551c88f9e309f4634c621"
             ],
             "index": "pypi",
+            "markers": "python_version >= '3.6'",
             "version": "==3.3.7"
         },
         "markdown-headdown": {
@@ -515,11 +512,11 @@
         },
         "setuptools": {
             "hashes": [
-                "sha256:35ab7fd3bcd95e6b7fd704e4a1539513edad446c097797f2985e0e4b960772f2",
-                "sha256:d59a21b17a275fb872a9c3dae73963160ae079f1049ed956880cd7c09b120538"
+                "sha256:f2504966861356aa38616760c0f66568e535562374995367b4e69c7143cf6bcd",
+                "sha256:fba5dd4d766e97be1b1681d98712680ae8f2f26d7881245f2ce9e40714f1a686"
             ],
             "markers": "python_version >= '3.8'",
-            "version": "==75.1.0"
+            "version": "==75.3.0"
         },
         "sgmllib3k": {
             "hashes": [
@@ -577,47 +574,47 @@
         },
         "watchdog": {
             "hashes": [
-                "sha256:0f9332243355643d567697c3e3fa07330a1d1abf981611654a1f2bf2175612b7",
-                "sha256:1021223c08ba8d2d38d71ec1704496471ffd7be42cfb26b87cd5059323a389a1",
-                "sha256:108f42a7f0345042a854d4d0ad0834b741d421330d5f575b81cb27b883500176",
-                "sha256:1e9679245e3ea6498494b3028b90c7b25dbb2abe65c7d07423ecfc2d6218ff7c",
-                "sha256:223160bb359281bb8e31c8f1068bf71a6b16a8ad3d9524ca6f523ac666bb6a1e",
-                "sha256:26dd201857d702bdf9d78c273cafcab5871dd29343748524695cecffa44a8d97",
-                "sha256:294b7a598974b8e2c6123d19ef15de9abcd282b0fbbdbc4d23dfa812959a9e05",
-                "sha256:349c9488e1d85d0a58e8cb14222d2c51cbc801ce11ac3936ab4c3af986536926",
-                "sha256:49f4d36cb315c25ea0d946e018c01bb028048023b9e103d3d3943f58e109dd45",
-                "sha256:53a3f10b62c2d569e260f96e8d966463dec1a50fa4f1b22aec69e3f91025060e",
-                "sha256:53adf73dcdc0ef04f7735066b4a57a4cd3e49ef135daae41d77395f0b5b692cb",
-                "sha256:560135542c91eaa74247a2e8430cf83c4342b29e8ad4f520ae14f0c8a19cfb5b",
-                "sha256:720ef9d3a4f9ca575a780af283c8fd3a0674b307651c1976714745090da5a9e8",
-                "sha256:752fb40efc7cc8d88ebc332b8f4bcbe2b5cc7e881bccfeb8e25054c00c994ee3",
-                "sha256:78864cc8f23dbee55be34cc1494632a7ba30263951b5b2e8fc8286b95845f82c",
-                "sha256:85527b882f3facda0579bce9d743ff7f10c3e1e0db0a0d0e28170a7d0e5ce2ea",
-                "sha256:90a67d7857adb1d985aca232cc9905dd5bc4803ed85cfcdcfcf707e52049eda7",
-                "sha256:91b522adc25614cdeaf91f7897800b82c13b4b8ac68a42ca959f992f6990c490",
-                "sha256:9413384f26b5d050b6978e6fcd0c1e7f0539be7a4f1a885061473c5deaa57221",
-                "sha256:94d11b07c64f63f49876e0ab8042ae034674c8653bfcdaa8c4b32e71cfff87e8",
-                "sha256:950f531ec6e03696a2414b6308f5c6ff9dab7821a768c9d5788b1314e9a46ca7",
-                "sha256:a2e8f3f955d68471fa37b0e3add18500790d129cc7efe89971b8a4cc6fdeb0b2",
-                "sha256:ae6deb336cba5d71476caa029ceb6e88047fc1dc74b62b7c4012639c0b563906",
-                "sha256:b8ca4d854adcf480bdfd80f46fdd6fb49f91dd020ae11c89b3a79e19454ec627",
-                "sha256:c66f80ee5b602a9c7ab66e3c9f36026590a0902db3aea414d59a2f55188c1f49",
-                "sha256:d52db5beb5e476e6853da2e2d24dbbbed6797b449c8bf7ea118a4ee0d2c9040e",
-                "sha256:dd021efa85970bd4824acacbb922066159d0f9e546389a4743d56919b6758b91",
-                "sha256:e25adddab85f674acac303cf1f5835951345a56c5f7f582987d266679979c75b",
-                "sha256:f00b4cf737f568be9665563347a910f8bdc76f88c2970121c86243c8cfdf90e9",
-                "sha256:f01f4a3565a387080dc49bdd1fefe4ecc77f894991b88ef927edbfa45eb10818"
+                "sha256:07df1fdd701c5d4c8e55ef6cf55b8f0120fe1aef7ef39a1c6fc6bc2e606d517a",
+                "sha256:20ffe5b202af80ab4266dcd3e91aae72bf2da48c0d33bdb15c66658e685e94e2",
+                "sha256:212ac9b8bf1161dc91bd09c048048a95ca3a4c4f5e5d4a7d1b1a7d5752a7f96f",
+                "sha256:2cce7cfc2008eb51feb6aab51251fd79b85d9894e98ba847408f662b3395ca3c",
+                "sha256:490ab2ef84f11129844c23fb14ecf30ef3d8a6abafd3754a6f75ca1e6654136c",
+                "sha256:6eb11feb5a0d452ee41f824e271ca311a09e250441c262ca2fd7ebcf2461a06c",
+                "sha256:6f10cb2d5902447c7d0da897e2c6768bca89174d0c6e1e30abec5421af97a5b0",
+                "sha256:7607498efa04a3542ae3e05e64da8202e58159aa1fa4acddf7678d34a35d4f13",
+                "sha256:76aae96b00ae814b181bb25b1b98076d5fc84e8a53cd8885a318b42b6d3a5134",
+                "sha256:7a0e56874cfbc4b9b05c60c8a1926fedf56324bb08cfbc188969777940aef3aa",
+                "sha256:82dc3e3143c7e38ec49d61af98d6558288c415eac98486a5c581726e0737c00e",
+                "sha256:9041567ee8953024c83343288ccc458fd0a2d811d6a0fd68c4c22609e3490379",
+                "sha256:90c8e78f3b94014f7aaae121e6b909674df5b46ec24d6bebc45c44c56729af2a",
+                "sha256:9513f27a1a582d9808cf21a07dae516f0fab1cf2d7683a742c498b93eedabb11",
+                "sha256:9ddf7c82fda3ae8e24decda1338ede66e1c99883db93711d8fb941eaa2d8c282",
+                "sha256:a175f755fc2279e0b7312c0035d52e27211a5bc39719dd529625b1930917345b",
+                "sha256:a1914259fa9e1454315171103c6a30961236f508b9b623eae470268bbcc6a22f",
+                "sha256:afd0fe1b2270917c5e23c2a65ce50c2a4abb63daafb0d419fde368e272a76b7c",
+                "sha256:bc64ab3bdb6a04d69d4023b29422170b74681784ffb9463ed4870cf2f3e66112",
+                "sha256:bdd4e6f14b8b18c334febb9c4425a878a2ac20efd1e0b231978e7b150f92a948",
+                "sha256:c7ac31a19f4545dd92fc25d200694098f42c9a8e391bc00bdd362c5736dbf881",
+                "sha256:c7c15dda13c4eb00d6fb6fc508b3c0ed88b9d5d374056b239c4ad1611125c860",
+                "sha256:c897ac1b55c5a1461e16dae288d22bb2e412ba9807df8397a635d88f671d36c3",
+                "sha256:cbafb470cf848d93b5d013e2ecb245d4aa1c8fd0504e863ccefa32445359d680",
+                "sha256:d1cdb490583ebd691c012b3d6dae011000fe42edb7a82ece80965b42abd61f26",
+                "sha256:e3df4cbb9a450c6d49318f6d14f4bbc80d763fa587ba46ec86f99f9e6876bb26",
+                "sha256:e6439e374fc012255b4ec786ae3c4bc838cd7309a540e5fe0952d03687d8804e",
+                "sha256:e6f0e77c9417e7cd62af82529b10563db3423625c5fce018430b249bf977f9e8",
+                "sha256:e7631a77ffb1f7d2eefa4445ebbee491c720a5661ddf6df3498ebecae5ed375c",
+                "sha256:ef810fbf7b781a5a593894e4f439773830bdecb885e6880d957d5b9382a960d2"
             ],
             "markers": "python_version >= '3.9'",
-            "version": "==5.0.3"
+            "version": "==6.0.0"
         },
         "whitenoise": {
             "hashes": [
-                "sha256:58c7a6cd811e275a6c91af22e96e87da0b1109e9a53bb7464116ef4c963bf636",
-                "sha256:a1ae85e01fdc9815d12fa33f17765bc132ed2c54fa76daf9e39e879dd93566f6"
+                "sha256:486bd7267a375fa9650b136daaec156ac572971acc8bf99add90817a530dd1d4",
+                "sha256:df12dce147a043d1956d81d288c6f0044147c6d2ab9726e5772ac50fb45d2280"
             ],
-            "markers": "python_version >= '3.8'",
-            "version": "==6.7.0"
+            "markers": "python_version >= '3.9'",
+            "version": "==6.8.2"
         },
         "zipp": {
             "hashes": [

```

which is the result of running

```
pipenv update "dc-django-utils@https://github.com/DemocracyClub/dc_django_utils/archive/refs/tags/8.0.0.tar.gz"
pipenv update "dc-design-system@https://github.com/DemocracyClub/design-system/archive/refs/tags/0.6.0.tar.gz"
```

(not `update` not `upgrade` )

against what is on main.

Comment (symroe) [reply Pipfile.lock]: Lol, I did the wrong one. I've now done the right one

Comment (chris48s) [reply Pipfile.lock]: Pipenv is a very difficult tool to use correctly.

Now your djhtml check is failing because you changed the formatting to account for the fact you'd upgraded djhtml (despite not intending to) but now you're back to the previous version. You'll also want to align your pre-commit with the lockfile again.

**Pull request merged**
